core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1409,6 +1426,77 @@ class KafkaApis(val requestChannel: RequestChannel,
       ))
   }
 
+  def handleDescribeTopicPartitionsRequest(request: RequestChannel.Request): Unit = {
```

Comment:
Can you move this code into a new class? KafkaApis is already much too large.

Reply:
Done.